### PR TITLE
Bug #164 Change AOS Product Images to use an entry point

### PR DESCRIPTION
**Pull request by shogunpol**
Recreate way of refer the image in Products module
## Description

- modify and add values in modules/AOS_Products/vardevs.php
- in modules/AOS_Products/AOS_Products delete old way of save the image
- in modules/AOS_Products/views/view.edit.php delete old way of display the image
- also few changes in other files (see sourcecode)
  References issue #164
  
  
  ## Motivation and Context
  
  Product images currently links directly to the image in the upload folder. This only works if SuiteCRM is configured poorly and the upload folder is accessible (i.e. the .htaccess files are being ignored). We should change the product image code to refer to the image via the download entry point.
  ## How To Test This
  
- Create Products records,
- Attach images into Products records,
- Open DetailView to see Small copy of picture
- Open EditView, check if you can download image
  ## Types of changes
  
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

**Comment by ijdavie** [comment]
@shogunpol - can you update travis-ci tests ?

**Comment by shogunpol** [comment]
@ijdavie , travis updated

File: tests/tests/modules/AOS_Products/AOS_ProductsTest.php [hunk 1]
```diff
@@ -28,14 +28,14 @@ public function testsave()
...
-        $_POST['deleteAttachment'] = '1';
+        $_POST['remove_file_product_image'] = '1';
 
         $aosProducts->save();
 
         //test for record ID to verify that record is saved
         $this->assertTrue(isset($aosProducts->id));
         $this->assertEquals(36, strlen($aosProducts->id));
-        $this->assertEquals('', $aosProducts->product_image);
+        $this->assertEquals('test img', $aosProducts->product_image);
```
Comment (samus-aran) [inline tests/tests/modules/AOS_Products/AOS_ProductsTest.php]: test img?

**Comment by horus68** [comment]
@Dillon-Brown is this PR lost in "changes requested"?

**Comment by Dillon-Brown** [comment]
@horus68 Looks like its just been missed. I'll see about remaking it for 7.8.x. Thanks!

File: tests/tests/modules/AOS_Products/AOS_ProductsTest.php (hunk 1 above)
Comment (Dillon-Brown) [reply tests/tests/modules/AOS_Products/AOS_ProductsTest.php]: $aosProducts->product_image = 'test img'; on line 30

**Comment by Dillon-Brown** [comment]
Assessed

**Pull request merged**